### ISSUE TITLE
Make labels optional

### ISSUE THREAD

**Issue by KevinMusgrave** [KevinMusgrave/pytorch-metric-learning#412]
If pairs/triplets are provided, labels are usually not used, yet they are still a required input to the loss function.
